test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsSolutionRestoreServiceTests.cs

```diff
@@ -1849,6 +1849,95 @@ public async Task NominateProjectAsync_CancelledToken_ThrowsOperationCanceledExc
+                            {
+                                new VsReferenceProperty("Version", "2.0.0")
+                            })),
+                            // the second centralPackageVersion with the same version name will be ignored 
+                            new VsReferenceItem("foo", new VsReferenceProperties(new []
+                            {
+                                new VsReferenceProperty("Version", "3.0.0")
+                            }))
+                        })
+            };
+
+            // Act
+            var result = VsSolutionRestoreService.ToPackageSpec(projectName, targetFrameworks, CommonFrameworks.NetStandard20.ToString(), string.Empty);
+
+            // Assert
+            var tfm = result.TargetFrameworks.First();
+
+            var packageVersion = Assert.Single(tfm.CentralPackageVersions);
+            Assert.Equal("foo", packageVersion.Key);
+            Assert.Equal("[2.0.0, )", packageVersion.Value.VersionRange.ToNormalizedString()) ;
+            var packageReference = Assert.Single(tfm.Dependencies);
+            Assert.Null(packageReference.LibraryRange.VersionRange);
+            Assert.True(result.RestoreMetadata.CentralPackageVersionsEnabled);
+        }
+
+        /// <summary>
+        /// The default for DisableCentralPackageVersions should be disabled.
+        /// </summary>
+        [Theory]
+        [InlineData("1.0.0", "false")]
+        [InlineData(null, null)]
+        [InlineData("1.0.0", "")]
+        public void ToPackageSpec_CentralVersions_AreNotAddedToThePackageSpecIfCPVMIsNotEnabled(string packRefVersion, string managePackageVersionsCentrally)
+        {
+            // Arrange
+            ProjectNames projectName = new ProjectNames(@"f:\project\project.csproj", "project", "project.csproj", "prjectC");
+            var emptyReferenceItems = Array.Empty<VsReferenceItem>();
+            var packageReferenceProperties = packRefVersion == null ?
+                new VsReferenceProperties() :
+                new VsReferenceProperties(new[] { new VsReferenceProperty("Version", packRefVersion) });
+            var projectProperties = managePackageVersionsCentrally == null ?
+                new VsProjectProperty[0] :
+                new VsProjectProperty[] { new VsProjectProperty("ManagePackageVersionsCentrally", managePackageVersionsCentrally) };
+
+            var targetFrameworks = new VsTargetFrameworkInfo3[] { new VsTargetFrameworkInfo3(
+                targetFrameworkMoniker: CommonFrameworks.NetStandard20.ToString(),
+                packageReferences: new[] { new VsReferenceItem("foo", packageReferenceProperties) },
+                projectReferences: emptyReferenceItems,
+                packageDownloads: emptyReferenceItems,
+                frameworkReferences: emptyReferenceItems,
+                projectProperties: projectProperties,
+                centralPackageVersions: new[]
```

Comment:
👍 I like that you're testing the versions being specified, without enabling the `ManagePackageVersionsCentrally` property.

Question❓🙋‍♂️:
If a user specifies these CPVM versions but does not enable them, will it be easy for them to realize their CPVM versions are not affecting anything? I am imagining that I forget to set CPVM enabled, but I add a huge list of Versions. Does something Validate this and tell me "hey, you have versions, but you forgot to enable CPVM" ?

Reply:
Currently there is not an warning. I will add an issue to track if we want to do anything specifically in this direction. Currently the min version will be used for PackageRefs.